hoth_api.yaml

```diff
@@ -319,5 +319,351 @@ components:
+        On the API side, the subtasks will filter through the provided required_services.service_offerings
+        to create the subtasks fields.
+      properties:
+        required_services:
+          type: array
+          items:
+            $ref: "#/components/schemas/RequiredService"
+          maxItems: 4000
+    RequiredService:
+      description: "Part of DOW, represents a single required service"
+      properties:
+        applicable_classification_levels:
+          type: array
+          items:
+            $ref: "#/components/schemas/ClassificationLevel"
+          maxItems: 4000 # This is the limit in SNOW, but it doesn't actually make sense to have a 4000 limit because there aren't 4000 classification levels
+        usage_description:
+          type: string
+          maxLength: 1000
+        need_for_entire_TO_duration:
+          type: boolean
+        applicable_periods:
+          type: array
+          items:
+            $ref: "#/components/schemas/Period"
+        service_offerings: # represents the "Select Service Offerings" field
+          type: array
+          items:
+            $ref: "#/components/schemas/ServiceOffering"
+        other_service_offering:
+          type: string
+          maxLength: 100
+    Period:
+      description: "Part of DOW, represents Period Table fields"
+      properties:
+        period_type:
+          type: string
+          enum:
+            - Base
+            - Option
+        period_unit_count:
+          type: integer
+        period_unit:
+          type: string
+          enum:
+            - Day(s)
+            - Week(s)
+            - Month(s)
+            - Year(s)
+        option_order:
+          type: integer
+    ServiceOffering:
+      description: "Part of DOW, represents a single Service Offering (Service Offering table)"
+      properties:
+        service_offering_group:
+          type: string
+          enum:
+            - Advanced Technology and Algorithmic Techniques (Machine Learning)
+            - Advisory and Assistance
+            - Applications
+            - Compute
+            - Database with Storage
+            - Developer Tools and Services
+            - Edge Computing and Tactical Edge (TE)
+            - General IaaS, PaaS, and SaaS
+            - Internet of Things (IoT)
+            - Networking
+            - Security
+            - Training
+        name:
+          type: string
+          maxLength: 100
+    PeriodOfPerformance:
+      description: "Part of DOW, represents a single Period of Performance"
+      properties:
+        base_period:
+          $ref: "#/components/schemas/Period"
+        option_periods:
+          type: array
+          items:
+            $ref: "#/components/schemas/Period"
+          maxItems: 4000
+        pop_start_request:
+          type: string
+          enum:
+            - Unselected
+            - Yes
+            - No
```

Comment:
Yeah let's not reinvent `enum Bool { True, False, FileNotFound }`  here [https://thedailywtf.com/articles/what_is_truth_0x3f_](https://thedailywtf.com/articles/what_is_truth_0x3f_)

Reply:
@juliusfitzhugh-ccpo That is accurate to my knowledge.  I'd only add that they did this because False was the default value and they wanted to avoid that in some cases.

Reply:
And the guiding principle for these objects that we're specifying here is that they will be populated _from_ SNOW data and then passed on to HOTH. In other words, it should be aimed at consumption by HOTH (and adapted if necessary, e.g. `"Unselected"` becomes `null`) rather than catering towards SNOW.